doc/redis-sentinel-tls.md

```diff
@@ -0,0 +1,688 @@
+You can store the certificates and keys in OpenShift as secrets for easy access.
+
+```bash
+oc create secret generic redis-tls-secret \
+  --from-file=redis-server.crt \
+  --from-file=redis-server.key \
+  --from-file=ca.crt
+
+
+oc create secret generic redis-sentinel-tls-secret \
+  --from-file=redis-sentinel.crt \
+  --from-file=redis-sentinel.key \
+  --from-file=ca.crt
+```
+
+These secrets will contain the necessary certificates for Redis and Sentinel.
+
+
+### Install Redis Server with TLS support
+
+#### Redis ConfigMap:
+
+```yaml
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config-redis  
+data:
+  redis.conf: |+
+    # redis.conf
+    bind 0.0.0.0
+    protected-mode no
+    port 6379
+    tls-port 6380
+    tls-cert-file /etc/redis/certs/redis-server.crt
+    tls-key-file /etc/redis/certs/redis-server.key
+    tls-ca-cert-file /etc/redis/certs/ca.crt
+    tls-auth-clients no
+    stop-writes-on-bgsave-error no
+    save ""
+EOF
+```
+
+#### Redis StatefulSet
+
+```yaml
+cat << EOF | oc create -f -
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: redis
+spec:
+  serviceName: "redis"
+  replicas: 3
+  selector:
+    matchLabels:
+      app: redis
+  template:
+    metadata:
+      labels:
+        app: redis
+    spec:
+      containers:
+      - name: redis
+        #image: quay.io/fedora/redis-6
+        image: redis:7
+        ports:
+          - containerPort: 6379
+          - containerPort: 6380
+        volumeMounts:
+          - name: redis-config-volume
+            mountPath: /etc/redis/redis.conf
+            subPath: redis.conf
+          - name: redis-tls-volume
+            mountPath: /etc/redis/certs
+            readOnly: true
+        command: ["/bin/sh", "-c", "redis-server /etc/redis/redis.conf"]
+      volumes:
+        - name: redis-config-volume
+          configMap:
+            name: redis-config-redis
+        - name: redis-tls-volume
+          secret:
+            secretName: redis-tls-secret
+EOF
+```
+
+
+### Install Sentinel with TLS support
+
+#### Sentinel Service
+
+```yaml
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-sentinel
+spec:
+  selector:
+    app: redis-sentinel
+  ports:
+    - name: redis 
+      protocol: TCP
+      port: 26379 
+      targetPort: 26379 
+    - name: redis-tls
+      protocol: TCP
+      port: 26380 
+      targetPort: 26380
+  clusterIP: None  # This makes the service headless
+EOF
+```
+
+#### Sentinel ConfigMap
+  
+```yaml
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-sentinel-config 
+data:
+  sentinel.conf: |
+    port 0
+    tls-port 26380
+    tls-cert-file /etc/redis/certs/redis-sentinel.crt
+    tls-key-file /etc/redis/certs/redis-sentinel.key
+    tls-ca-cert-file /etc/redis/certs/ca.crt
+    tls-auth-clients yes 
+    tls-replication yes
+
+    #sentinel monitor mymaster redis.3scale-test.svc.cluster.local 6380 2
```

Comment:
might be worth to explain that this is using the DNS instead and is recommended. My ns is different so maybe something like "sentinel monitor mymaster redis.<NAMESPACE>.svc.cluster.local 6380 2"

Reply:
I've tried it with dns and it failed.

Reply:
I tried more but didn't succed to work with DNS for master monitoring . 
New section added : 
![image](https://github.com/user-attachments/assets/dfe6fa8b-a0b8-49ed-9b3a-7cb38013e68b)